nifi-nar-bundles/nifi-html-bundle/nifi-html-processors/src/test/java/org/apache/nifi/TestGetHTMLElement.java

```diff
@@ -0,0 +1,319 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.util.MockFlowFile;
+import org.apache.nifi.util.TestRunner;
+import org.apache.nifi.util.TestRunners;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.lang.Exception;
+import java.util.List;
+
+import static org.junit.Assert.assertTrue;
+
+public class TestGetHTMLElement extends AbstractHTMLTest {
+
+    private TestRunner testRunner;
+
+    @Before
+    public void init() {
+        testRunner = TestRunners.newTestRunner(GetHTMLElement.class);
+        testRunner.setProperty(GetHTMLElement.URL, "http://localhost");
+        testRunner.setProperty(GetHTMLElement.OUTPUT_TYPE, GetHTMLElement.ELEMENT_HTML);
+        testRunner.setProperty(GetHTMLElement.DESTINATION, GetHTMLElement.DESTINATION_CONTENT);
+        testRunner.setProperty(GetHTMLElement.HTML_CHARSET, "UTF-8");
+    }
+
+    @Test
+    public void testNoElementFound() throws Exception {
+        testRunner.setProperty(GetHTMLElement.CSS_SELECTOR, "b");   //Bold element is not present in sample HTML
+//        testRunner.setProperty(GetHTMLElement.APPEND_ELEMENT_VALUE, "");
+
+        ProcessSession session = testRunner.getProcessSessionFactory().createSession();
+        FlowFile ff = writeContentToNewFlowFile(HTML.getBytes(), session);
+
+        testRunner.enqueue(ff);
+        testRunner.run();
+
+        testRunner.assertTransferCount(GetHTMLElement.REL_SUCCESS, 0);
+        testRunner.assertTransferCount(GetHTMLElement.REL_FAILURE, 0);
+        testRunner.assertTransferCount(GetHTMLElement.REL_NOT_FOUND, 1);
+    }
+
+    @Test
+    public void testInvalidSelector() throws Exception {
+        testRunner.setProperty(GetHTMLElement.CSS_SELECTOR, "InvalidCSSSelectorSyntax");
+
+        ProcessSession session = testRunner.getProcessSessionFactory().createSession();
+        FlowFile ff = writeContentToNewFlowFile(HTML.getBytes(), session);
+
+        testRunner.enqueue(ff);
```

Comment:
This all is a lot easier if we just do testRunner.enqueue(HTML.getBytes()) - no need to create a session and write to a FlowFile.